libs/template/src/lib/push/push.pipe.ts

```diff
@@ -76,61 +93,60 @@ export class PushPipe<U> implements PipeTransform, OnDestroy {
     this.strategyProvider.primaryStrategy,
     this.strategyProvider.strategies
   );
+  /** @internal */
+  private patchZone: false | NgZone;
+  /** @internal */
+  private _renderCallback: NextObserver<U>;
 
   constructor(
     private strategyProvider: RxStrategyProvider,
-    private cdRef: ChangeDetectorRef
+    private cdRef: ChangeDetectorRef,
+    private ngZone: NgZone,
+    @Optional()
+    @Inject(RX_ANGULAR_CONFIG)
+    private config?: RxAngularConfig<S>
   ) {
-    const scope = (cdRef as any).context;
-    this.subscription = this.templateObserver.values$
-      .pipe(
-        filter<RxNotification<U>>(
-          (n) =>
-            n.kind === RxNotificationKind.suspense ||
-            n.kind === RxNotificationKind.next
-        ),
-        tap<RxNotification<U>>((notification) => {
-          this.renderedValue = notification.value as U;
-        }),
-        withLatestFrom(this.strategyHandler.strategy$),
-        switchMap(([v, strategy]) =>
-          this.strategyProvider.schedule(
-            () => {
-              strategy.work(cdRef, scope);
-            },
-            {
-              scope,
-              strategy: strategy.name,
-              patchZone: this.strategyProvider.config.patchZone ? null : false,
-            }
-          )
-        )
-      )
-      .subscribe();
+    this.subscription = this.handleChangeDetection();
   }
 
   transform(
     potentialObservable: null,
-    config?: string | Observable<string>,
+    config?: RxStrategyNames<S> | Observable<RxStrategyNames<S>>,
     renderCallback?: NextObserver<U>
   ): null;
   transform(
     potentialObservable: undefined,
-    config?: string | Observable<string>,
+    config?: RxStrategyNames<S> | Observable<RxStrategyNames<S>>,
     renderCallback?: NextObserver<U>
   ): undefined;
   transform(
     potentialObservable: ObservableInput<U>,
-    config?: string | Observable<string>,
+    config?: RxStrategyNames<S> | Observable<RxStrategyNames<S>>,
     renderCallback?: NextObserver<U>
   ): U;
+  transform(
+    potentialObservable: ObservableInput<U>,
+    config?: RxComponentInput<U, S>
+  ): U;
   transform(
     potentialObservable: ObservableInput<U> | null | undefined,
-    config: string | Observable<string> | undefined,
+    config:
+      | RxComponentInput<U, S>
+      | RxStrategyNames<S>
+      | Observable<RxStrategyNames<S>>
+      | undefined,
     renderCallback?: NextObserver<U>
   ): U | null | undefined {
+    this._renderCallback = renderCallback;
     if (config) {
-      this.strategyHandler.next(config);
+      if (isRxComponentInput(config)) {
+        this.strategyHandler.next(config.strategy as string);
+        this._renderCallback = config.renderCallback;
+        // set fallback if patchZone is not set
+        this.setPatchZone(config.patchZone);
```

Comment:
Properties from `config` object could be possibly undefined, should these setters be conditionally wrapped to avoid passing undefined?

Reply:
strategyHandler internally takes care of falling back to defaultStrategy when a falsy value is set. Since we do not subscribe to renderCallback at all it's fine to just overwrite it.
Anything else I am missing here?